<?xml version="1.0" encoding="UTF-8"?>
<launch>
	<arg name="initial_pose_in_map_to_base" default="true" />
	<arg name="initial_x" default="0.0" />
	<arg name="initial_y" default="0.0" />
	<arg name="initial_z" default="0.0" />
	<arg name="initial_roll" default="0.0" />
	<arg name="initial_pitch" default="0.0" />
	<arg name="initial_yaw" default="0.0" />
	<arg name="pose_with_covariance_stamped_topic" default="/initialpose" />
	<arg name="pose_stamped_topic" default="" />
	<arg name="odometry_topic" default="" />
	<arg name="map_frame_id" default="map" />
	<arg name="odom_frame_id" default="odom" />
	<arg name="publish_rate" default="100.0" />
	<arg name="publish_last_pose_tf_timeout_seconds" default="-1.0" />
	<arg name="base_link_frame_id" default="base_footprint" />
	<arg name="invert_tf_transform" default="false" />
	<arg name="invert_tf_hierarchy" default="false" />
	<arg name="transform_pose_to_map_frame_id" default="true" />
	<arg name="parameters_namespace" default="dynamic_robot_localization" />


	<!-- tf publisher for localization pose -->	
	<node pkg="dynamic_robot_localization" type="drl_pose_to_tf_publisher_node" name="drl_pose_to_tf_publisher_node" ns="$(arg parameters_namespace)" output="screen">
		<param name="publish_rate" type="double" value="$(arg publish_rate)" />
		<param name="publish_last_pose_tf_timeout_seconds" type="double" value="$(arg publish_last_pose_tf_timeout_seconds)" /> <!-- if < 0 last pose received will be published forever with no timeout (until a new pose is received and updates the current publish pose) -->
		<param name="pose_with_covariance_stamped_topic" type="str" value="$(arg pose_with_covariance_stamped_topic)" />
		<param name="pose_stamped_topic" type="str" value="$(arg pose_stamped_topic)" />
		<param name="odometry_topic" type="str" value="$(arg odometry_topic)" />
		<param name="map_frame_id" type="str" value="$(arg map_frame_id)" />
		<param name="odom_frame_id" type="str" value="$(arg odom_frame_id)" />
		<param name="base_link_frame_id" type="str" value="$(arg base_link_frame_id)" />
		<param name="invert_tf_transform" type="bool" value="$(arg invert_tf_transform)" />
		<param name="invert_tf_hierarchy" type="bool" value="$(arg invert_tf_hierarchy)" />
		<param name="transform_pose_to_map_frame_id" type="bool" value="$(arg transform_pose_to_map_frame_id)" />
		<param name="initial_pose_in_map_to_base" type="bool" value="$(arg initial_pose_in_map_to_base)" />
		<param name="initial_x" type="double" value="$(arg initial_x)" />
		<param name="initial_y" type="double" value="$(arg initial_y)" />
		<param name="initial_z" type="double" value="$(arg initial_z)" />
		<param name="initial_roll" type="double" value="$(arg initial_roll)" />
		<param name="initial_pitch" type="double" value="$(arg initial_pitch)" />
		<param name="initial_yaw" type="double" value="$(arg initial_yaw)" />
	</node>
</launch>